### PR TITLE
The API now also supports updating a subprofile.

### PR DESCRIPTION
**Pull request by jbreuer**
It's now possible to also update a subprofile with the API. It has some breaking changes because the CopernicaSubprofile and ICopernicaSubprofile inherit from CopernicaProfile and ICopernicaProfile. So maybe we can release this as v2?

### CODE REVIEW

**Comment by LennardF1989** [comment]
Just peaking at the changes, but things look great Jeroen, thanks! I'll ask @Atzie to give this a test-run ASAP :)

**Comment by jbreuer** [comment]
Thanks :-). I'll probably have some more improvements tomorrow.

**Comment by jbreuer** [comment]
Please don't merge this yet. I've had some feedback from Copernica and there are some better APIs that I can use. Will update this PR.

**Comment by jbreuer** [comment]
I've added a new commit to this PR. Now the API doesn't just support updating a subprofile, but also a CreateOrUpdate for both a profile and subprofile. When you use the CreateOrUpdate methods it will check if a profile or subprofile already exits. If it does it will update and if it doesn't it will create a new one. This way I only need to use 1 method to send my data to Copernica and the system will know what to do with it. It uses some new API features which Copernica informed me about.

The Copernica API URL is now also a setting. Copernica has a dev API (https://api-dev.copernica.com/) which sometimes is easy for testing new features. This way it's easy to use that API.

**Comment by jbreuer** [comment]
Any idea if/when this PR can be merged? I'm currently using my own version at https://github.com/jbreuer/CopernicaNET by referencing the dll instead of using NuGet. So far I haven't run into any issues.

**Comment by ynzenu** [comment]
Thank you for submitting the PR. I will check it out this afternoon.

File: Src/CopernicaNET.Sample/Web.config [hunk 1]
```diff
@@ -3,10 +3,10 @@
   <configSections>
     <section name="copernicanet" type="Arlanet.CopernicaNET.Configuration.CopernicaSettings"/>
   </configSections>
-  <copernicanet accesstoken="replacewithaccesstoken">
+  <copernicanet accesstoken="9cce4d325168bbf208bc9fc835151130c5349a5c61ecdf21c4a921df18bf1eca293d129d23c7bdc1d20982e2c5d0f0a0a6796afa473de276ccf9f83e315e54ea">
```
Comment (ynzenu) [inline Src/CopernicaNET.Sample/Web.config]: This is your accesstoken? Don't think you want this in the test project.

**Comment by ynzenu** [comment]
Alright your pull request has been merged. The nuget package will be updated later today/tonight.

**Comment by jbreuer** [comment]
Thanks for merging this. I think the nuget package version should be 2.0 because I made some changes to interfaces which could be breaking changes.

**Pull request merged**